pkg/prober/prober_windows.go

```diff
@@ -0,0 +1,108 @@
+
+	}
+	// win32 reference: https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certopensystemstorea
+	// If the function succeeds, it returns a handle to the specified certificate store.
+	storeHandle, err := syscall.CertOpenSystemStore(0, root)
+	if err != nil {
+		return nil, fmt.Errorf("[GetSystemCertPoolWindows] unable to open system certContext store: %v", syscall.GetLastError())
+	}
+
+	var certs []*x509.Certificate
+	var certContext *syscall.CertContext
+
+	// ref for why flags value is 0: https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certclosestore
+	defer func(store syscall.Handle, flags uint32) {
+		_ = syscall.CertCloseStore(store, flags)
+	}(certContext.Store, 0)
+
+	for {
+		certContext, err = syscall.CertEnumCertificatesInStore(storeHandle, certContext)
+		if err != nil {
+			if errno, ok := err.(syscall.Errno); ok {
+				if errno == CRYPT_E_NOT_FOUND {
+					// the for loop + the syscall for `CertEnumCertificatesInStore` will iterate through
+					// all available certContexts in the specified certificate store and
+					// returns a single certContext containing all certificates in the store
+					// if the error returned here is CRYPTO_E_NOT_FOUND, that indicates no certificates were found.
+					// This happens if the store is empty or if the function reached the end of the store's list.
+					// ref: https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certenumcertificatesinstore#return-value
+					logrus.Debugf("[GetSystemCertPoolWindows] no certificates were returned from the root CA store for probe (%s), expected a non-empty return value", probeName)
+					break
+				}
+			}
+			logrus.Errorf("[GetSystemCertPoolWindows] unable to enumerate certs in system certContext store for probe (%s): %v", probeName, syscall.GetLastError())
+		}
+		if certContext == nil {
+			logrus.Errorf("[GetSystemCertPoolWindows] certificate context returned from syscall is nil for probe (%s)", probeName)
+			break
+		}
+
+		// buf is a ~1048 kilobyte array that serves as a buffer holding the encoded value
+		// for each CA certificate in the Windows root CA store
+		// we use a binary shift to create a ~1048 Kb buffer (slightly larger than 1 megabyte)
+		// [1 << 20]byte -> (1*2)^20 = 1048576 bytes
+		// maximum size of a Windows certificate store is 16 kilobytes and is not related to number of certificates
+		// (1048576*8)/4096 = 2048 (amount of 4096-bit certificates that can be stored)
+		buf := (*[1 << 20]byte)(unsafe.Pointer(certContext.EncodedCert))[:]
+
+		// buf2 is an array of bytes equal to the length of the certContext pointer
+		// which contains a certificate from the Root CA store
+		buf2 := make([]byte, certContext.Length)
+		copy(buf2, buf)
+
+		// validate the root CA certificate and return a x509.Certificate pointer
+		// that is appended into our array of x509 certificates
+		c, err := x509.ParseCertificate(buf2)
```

Comment:
Can we perhaps just slice this to the correct length, instead of allocating a second slice of the correct length and then copying the bytes over?

```suggestion
		buf := (*[1 << 20]byte)(unsafe.Pointer(certContext.EncodedCert))[:certContext.Length]

		// validate the root CA certificate and return a x509.Certificate pointer
		// that is appended into our array of x509 certificates
		c, err := x509.ParseCertificate(buf)
```

Reply:
I wasn't sure how to do this but I knew there was a better way. thank you @brandond !